code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -43,6 +43,9 @@
 		// Changeling egg can survive in aliens!
 		var/mob/living/carbon/C = target
 		if(C.stat == DEAD)
+			if(is_type_in_list(/obj/item/organ/internal/body_egg, C.GetAllContents()))
```

Comment:
Yeah. It's probably intended since both alien and changeling egg are parasites medically wise.